packages/playwright-test/src/docker/docker.ts

```diff
@@ -129,25 +131,32 @@ export async function buildPlaywrightImage() {
 }
 
 export async function configureTestRunnerToUseDocker() {
-  console.log(colors.dim('Using docker container to run browsers.'));
-  await checkDockerEngineIsRunningOrDie();
-  let info = await containerInfo();
-  if (!info) {
-    process.stdout.write(colors.dim(`Starting docker container... `));
-    const time = Date.now();
-    info = await ensurePlaywrightContainerOrDie();
-    const deltaMs = (Date.now() - time);
-    console.log(colors.dim('Done in ' + (deltaMs / 1000).toFixed(1) + 's'));
-    console.log(colors.dim('The Docker container will keep running after tests finished.'));
-    console.log(colors.dim('Stop manually using:'));
-    console.log(colors.dim('    npx playwright docker stop'));
-  }
-  console.log(colors.dim(`View screen: ${info.vncSession}`));
-  process.env.PW_TEST_CONNECT_WS_ENDPOINT = info.wsEndpoint;
-  process.env.PW_TEST_CONNECT_HEADERS = JSON.stringify({
-    'x-playwright-proxy': '*',
-  });
-  process.env.PLAYWRIGHT_DOCKER = '1';
+  addRunnerPlugin(() => ({
```

Comment:
Perhaps we should add the plugin unconditionally, and just check the env variable here? This way `cli.ts` could probably be simplified.